src/libsyntax/attr.rs

```diff
@@ -356,25 +358,79 @@ pub fn requests_inline(attrs: &[Attribute]) -> bool {
     }
 }
 
+#[derive(Debug, Eq, PartialEq)]
+pub struct GatedCfg {
+    span: Span,
+    kind: GatedCfgKind,
+}
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Copy, Clone)]
+enum GatedCfgKind {
+    TargetFeature
+}
```

Comment:
Could this be refactored to instead be an array? It's convenient that when modifying `feature_gate.rs` that you only have to add an entry to a list instead of updating a few separate portions of code. We may grow some other gated cfg matchers here later so it'd be good to not have to spray logic in too many different places.